packages/venia-concept/src/actions/user/asyncActions.js

```diff
@@ -73,9 +73,25 @@ export const createAccount = accountInfo =>
             dispatch(assignGuestCartToCustomer());
         } catch (error) {
             dispatch(actions.createAccountError.receive(error));
+
+            /*
+             * Throw error again to notify async action which dispatched handleCreateAccount.
+             */
+            throw error;
```

Comment:
James, in general, redux-thunk breaks flux architecture, it is the idea of the package. In Flux architecture actions can only be a plain objects and action creator should return the object.
In this case, in my opinion, it is OK to use this approach to chain redux-thunk actions, it better thatn passing some callbacks to redux-thunk action. In many situations we need to know the result of executing await dispatch(someAsyncAction()); just to execute the code below this line only in case of successful someAsyncAction